Sources/SwiftDocC/Model/Rendering/Symbol/AvailabilityRenderMetadataItem.swift

```diff
@@ -11,7 +11,7 @@
 import Foundation
 import SymbolKit
 
-extension SymbolGraph.SemanticVersion {
+extension VersionTriplet {
```

Comment:
The reason for it is that we don't always have a `SymbolGraph.SemanticVersion` type when we want to convert the semantic version to a string representation, for example here `availability.introducedVersion` is a `VersionTriplet`:
https://github.com/anferbui/swift-docc/blob/f5f320b75efe8da4af9198a0b11b381a6d6cc3c4/Sources/SwiftDocC/Model/Rendering/Symbol/AvailabilityRenderMetadataItem.swift#L143

I think it makes more sense to have the string representation function as part of `VersionTriplet` than `SymbolGraph.SemanticVersion` since `VersionTriplet` is the type that docc uses internally, WDYT?

`VersionTriplet` is also how we store the current platforms so it makes more sense to me to use that as the type for comparing semantic versions and deriving a string representation.

Reply:
There's some overlap here with [this comment](https://github.com/apple/swift-docc/pull/959#discussion_r1658866010) but it seems we have many different version types in Swift DocC. 

- `VersionTriplet` is created in 2 places and used as a stored property in 1 place
- `Version` is used in 3 places (2 of which uses it to create a `VersionTriplet`) and used as a stored property in 1 internal place
- `Platform.Version` used as a stored property in 1 place
- `SemanticVersion` is used as a stored property in 5 places.

Additionally, `SymbolGraph.SemanticVersion` is used a few places that operate on other types from SymbolKit.

Considering that `SemanticVersion` and `SymbolGraph.SemanticVersion` contain the same information and behaves the same, maybe it makes more sense to move towards using that type over `VersionTriplet`. What do you think?